packages/plugins/Gfi/src/store/actions/debouncedGfiRequest.ts

```diff
@@ -0,0 +1,159 @@
+} from 'geojson'
+import { PolarActionContext } from '@polar/lib-custom-types'
+import { Feature } from 'ol'
+import { Geometry } from 'ol/geom'
+import VectorLayer from 'ol/layer/Vector'
+import { addFeature } from '../../utils/displayFeatureLayer'
+import { requestGfi } from '../../utils/requestGfi'
+import sortFeatures from '../../utils/sortFeatures'
+import { GfiGetters, GfiState } from '../../types'
+
+const mapFeaturesToLayerIds = (
+  {
+    getters: { layerKeys },
+    rootGetters: { configuration },
+  }: PolarActionContext<GfiState, GfiGetters>,
+  features: (symbol | GeoJsonFeature<GeoJsonGeometry, GeoJsonProperties>[])[],
+  srsName: string
+) => {
+  const generalMaxFeatures: number =
+    configuration.gfi?.maxFeatures || Number.POSITIVE_INFINITY
+  const featuresByLayerId = layerKeys.reduce(
+    (accumulator, key, index) => ({
+      ...accumulator,
+      [key]: Array.isArray(features[index])
+        ? (features[index] as []).slice(
+            0,
+            configuration.gfi?.layers[key].maxFeatures || generalMaxFeatures
+          )
+        : features[index],
+    }),
+    {}
+  )
+  return Object.entries(featuresByLayerId).reduce(
+    (accumulator, [layerKey, layerValues]) => ({
+      ...accumulator,
+      [layerKey]:
+        Array.isArray(layerValues) && layerValues.length >= 2
+          ? layerValues.sort((a, b) => sortFeatures(a, b, srsName))
+          : layerValues,
+    }),
+    {}
+  )
+}
+
+const getPromisedFeatures = (
+  {
+    rootGetters: { map, configuration },
+    getters: { layerKeys },
+  }: PolarActionContext<GfiState, GfiGetters>,
+  coordinate: [number, number]
+) =>
+  layerKeys.map((key) => {
+    const layer = map
+      .getLayers()
+      .getArray()
+      .find((layer) => layer.getProperties().id === key)
+
+    if (!layer) {
+      console.error(
+        `@polar/plugin-gfi: No layer with id "${key}" found during run-time. GFI skipped.`
+      )
+      return [] as GeoJsonFeature[]
+    }
+
+    const layerConfiguration = configuration.gfi?.layers[key] || {}
+    const layerSpecification = rawLayerList.getLayerWhere({ id: key })
+    const mainLayerConfiguration = configuration.layers.find(
+      (element) => element.id === key
+    )
+    const layerGfiMode =
+      mainLayerConfiguration?.gfiMode || configuration?.gfi?.mode || 'bboxDot'
+
+    return requestGfi({
+      map,
+      layer,
+      coordinate,
+      layerConfiguration,
+      layerSpecification,
+      mode: layerGfiMode,
+    })
+  })
+
+const filterFeatures = (
+  featuresByLayerId: Record<
+    string,
+    symbol | GeoJsonFeature<GeoJsonGeometry, GeoJsonProperties>[]
+  >
+): Record<string, GeoJsonFeature<GeoJsonGeometry, GeoJsonProperties>[]> => {
+  const entries = Object.entries(featuresByLayerId)
+  const filtered = entries.filter((keyValue) => Array.isArray(keyValue[1])) as [
+    string,
+    GeoJsonFeature<GeoJsonGeometry, GeoJsonProperties>[]
+  ][]
+  return Object.fromEntries(filtered)
+}
```

Comment:
This is new. Care to elaborate?

Reply:
This is filtering out pairs with symbol as value. The `afterLoadFunction` does not accept `symbol`s as part of the input. Practically, this never seems to have occured. Theoretically, it's possible; iirc if a subset of requests fail.